src/main/java/org/pkcs11/jacknji11/AttributeLengthStrategy.java

```diff
@@ -41,23 +41,26 @@ public interface AttributeLengthStrategy {
      * <p>
      * Use to avoid querying length of attributes for every call to C_GetAttributeValue.
      * <p>
-     * Note that implementations need to adhere strictly to the PKCS#11 for this strategy 
-     * to work. One detail where implementations (even popular ones) are known to be
-     * non-compliant is returning CKR_BUFFER_TOO_SMALL and setting uLVlaueLength to 
-     * CK_UNAVAILABLE_INFORMATION if sent a pre-allocated but too small buffer.
-     * Due to this, using this strategy requires careful testing with your HSM, for 
-     * your use case.
-     *
+     * Note that implementations need to adhere strictly to PKCS#11 for this strategy 
+     * to work robustly. One detail that this implementation relies on is that PKCS#11, 
+     * base specification section 5.7.5 C_GetAttributeValue,
+     * specifies returning CKR_BUFFER_TOO_SMALL and setting uLValueLen to 
+     * CK_UNAVAILABLE_INFORMATION, if sent a pre-allocated buffer that is too small to hold the value.
+     * There are multuple variations of implementation errors with either the wrong return value
+     * or failing to set uLValueLen to the correct value.
+     * Due to these common implementation issued, using this strategy requires careful testing 
+     * with your HSM, for your use case.
      */
     class MaxLengthStrategy implements AttributeLengthStrategy {
 
         /**
          * Default of 2KB has been established by following facts:
          * Modulus of 15Kb RSA (maximum) is around 2KB.
-         * For 7168Kb RSA (which is a practical limit) certificates are about 2K (without extensions).
+         * For 8192 bit RSA (which is a practical limit) certificates are just above about 2K
```

Comment:
Originally I was trying to explain why 2KB is enough.

Now, instead of explaining that 2KB is not enough for some uses then maybe we should change the default so that it is enough for practical limits. Like 2.5KB.

Reply:
It's a balance of course, Small enough to not pose unnecessary overhead, but large enough to speed up common use cases. Imho 8K RSA certificates (public keys fit on the 2KB limit) is a corner case.
A bigger issue in practice, for my side, is that CKA_SUBJECT or CKA_ID would be larger than 72 bytes.

Reply:
Modified the text slightly again, with more reasoning around the limit.